src/java.desktop/windows/classes/sun/awt/windows/ThemeReader.java

```diff
@@ -103,18 +103,8 @@ private static Long getThemeImpl(String widget, int dpi) {
                 theme = openTheme(widget, dpi);
             }
 
-            if(dpiAwareWidgetToTheme.containsKey(dpi))
-            {
-                if (!dpiAwareWidgetToTheme.get(dpi).isEmpty())
-                {
-                    dpiAwareWidgetToTheme.get(dpi).put(widget, theme);
-                }
-            }
-            else
-            {
-                dpiAwareWidgetToTheme.put(dpi, new HashMap<>());
-                dpiAwareWidgetToTheme.get(dpi).put(widget, theme);
-            }
+            dpiAwareWidgetToTheme.computeIfAbsent(dpi, key->new HashMap<>()).put(widget, theme);
```

Comment:
```suggestion
            dpiAwareWidgetToTheme.computeIfAbsent(dpi, key -> new HashMap<>()).put(widget, theme);
```
There are usually spaces around `->` in lambda expressions.